shadowsocks/tcp.go

```diff
@@ -253,14 +268,21 @@ func (s *tcpService) handleConnection(listenerPort int, clientConn onet.DuplexCo
 			logger.Debugf(status+": %v in %s sent %d bytes", clientConn.RemoteAddr(), clientLocation, proxyMetrics.ClientProxy)
 			return onet.NewConnectionError(status, "Replay detected", nil)
 		}
-		// Clear the authentication deadline
-		clientConn.SetReadDeadline(time.Time{})
 
 		ssr := NewShadowsocksReader(clientReader, cipherEntry.Cipher)
+		tgtAddr, err := socks.ReadAddr(ssr)
+		if err != nil {
+			// Drain to prevent a close on cipher error.
+			clientConn.(io.WriterTo).WriteTo(ioutil.Discard)
```

Comment:
Do you think this is likely a probe?  If so, we should probably use `absorbProbe`.